zephyr/lib/alloc.c

```diff
@@ -25,7 +25,9 @@
 #include <zephyr/pm/policy.h>
 #include <version.h>
 #include <zephyr/sys/__assert.h>
+#ifndef CONFIG_IMX93_A55
```

Comment:
Hmm, this is a bit iffy, but I don't see existing include guards to check whether soc.h exists for a platform or not, so maybe this is the pragmatic option now. We we'll have more platforms, we need to come up with something cleaner.

Reply:
Agreed. This doesn't scale very well. This should only be temporary until `soc.h` is introduced (probably when drivers are ported to Zephyr).

Reply:
ACK, since there's already 3 people who believe introducing `soc.h` would be a better idea I will submit a Zephyr PR and mark this as a draft until that one gets merged.

Reply:
we should be able to remove this include, not sure why is it needed here, did you try to remove it altogether?

Reply:
> we should be able to remove this include, not sure why is it needed here, did you try to remove it altogether?

Thanks for the suggestion ! I've tried removing all `soc.h` occurrences and it seems fine. Updated PR.

Reply:
Even better this way, looks good.